### PR TITLE
Use Arrays#mismatch for Outputs#common operations

### PR DESCRIPTION
**Pull request by gf2121**
Make `Outputs#common` take advantage of `Arrays#mismatch`.

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java [hunk 1]
```diff
@@ -43,28 +44,29 @@ public IntsRef common(IntsRef output1, IntsRef output2) {
...
+            output2.ints,
+            output2.offset,
+            output2.offset + output2.length);
 
-    if (pos1 == output1.offset) {
+    if (mismatchPos == 0) {
       // no common prefix
       return NO_OUTPUT;
-    } else if (pos1 == output1.offset + output1.length) {
+    } else if (mismatchPos == -1) {
```
Comment (dungba88) [inline lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java:59]: Why aren't we merging these 2 branches?

```
else if (mismatchPos == -1 || mismatchPos == output1.length)
```

File: lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java [hunk 2]
```diff
@@ -43,28 +44,29 @@ public IntsRef common(IntsRef output1, IntsRef output2) {
...
+    } else if (mismatchPos == output1.length) {
       // output1 is a prefix of output2
       return output1;
-    } else if (pos2 == output2.offset + output2.length) {
+    } else if (mismatchPos == output2.length) {
       // output2 is a prefix of output1
       return output2;
     } else {
-      return new IntsRef(output1.ints, output1.offset, pos1 - output1.offset);
+      return new IntsRef(output1.ints, output1.offset, mismatchPos);
```
Comment (dungba88) [inline lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java:69]: I think the 3rd parameter is length, so it should be `mismatchPos - output1.offset` instead?

File: lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java (hunk 1 above)
Comment (gf2121) [reply lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java:59]: Just for better comment to explain. Will merge if you think this way is better :)

**Review by jpountz — APPROVED** [review]

File: lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java (hunk 1 above)
Comment (dungba88) [reply lucene/core/src/java/org/apache/lucene/util/fst/IntSequenceOutputs.java:59]: Either way would look good to me.

**Review by dungba88 — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Pull request merged**